client/client.go

```diff
@@ -29,48 +31,59 @@ type PolicyConfig struct {
 	}
 }
 
+type QueryResult struct {
+	Name          string
+	CheckPassed   bool
+	ResultHeaders []string
+	ResultRows    [][]string
+}
+
+
 type Client struct {
 	driver  string
 	dsn     string
-	verbose string
 	db      *database.Database
-	runSelf bool
-	config  Config
+	// config is the client's configuration regarding all providers, client details etc'
+	config  *config.Config
+	// access to CloudQuery plugin hub
+	hub hub.Hub
 }
 
-func New(driver string, dsn string) (*Client, error) {
-	client := Client{
-		driver:  driver,
-		dsn:     dsn,
+func New(configPath, driver string, dsn string) (Client, error) {
```

Comment:
why you decided to not return an address of. This way you can pass `nil` and not an un-initialised object.